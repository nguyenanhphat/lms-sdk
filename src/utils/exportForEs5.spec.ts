describe('Exported functions', () => {
  test('Exported default function', () => {
    const { default: exportForEs5 } = require('./exportForEs5');
    expect(exportForEs5).toBeInstanceOf(Function);
  });
});

describe('Transform is code', () => {
  const SAMPLES = [
    [undefined, undefined],
    [null, null],
    ['empty', ''],
    ['function', () => {}], // tslint:disable-line:no-empty
    ['string', '__string__'],
    ['number', 123],
    ['object', { key: 'value' }],
    ['map', new Map([['key', 'value']])],
    ['set', new Set(['key1', 'key2'])],
  ];

  const { default: exportForEs5 } = require('./exportForEs5');
  test.each(SAMPLES)('Type is %j', (type: string, code: any) => {
    expect(exportForEs5(code)).toMatchSnapshot();
  });
});
