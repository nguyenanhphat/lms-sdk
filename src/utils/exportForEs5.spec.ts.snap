// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Transform is code Type is "empty" 1`] = `
"
  \\"use strict\\";
  Object.defineProperty(exports, \\"__esModule\\", {
      value: true
  });
  exports.default = \\"\\";
"
`;

exports[`Transform is code Type is "function" 1`] = `
"
  \\"use strict\\";
  Object.defineProperty(exports, \\"__esModule\\", {
      value: true
  });
  exports.default = function() {};
"
`;

exports[`Transform is code Type is "map" 1`] = `
"
  \\"use strict\\";
  Object.defineProperty(exports, \\"__esModule\\", {
      value: true
  });
  exports.default = {};
"
`;

exports[`Transform is code Type is "number" 1`] = `
"
  \\"use strict\\";
  Object.defineProperty(exports, \\"__esModule\\", {
      value: true
  });
  exports.default = 123;
"
`;

exports[`Transform is code Type is "object" 1`] = `
"
  \\"use strict\\";
  Object.defineProperty(exports, \\"__esModule\\", {
      value: true
  });
  exports.default = {\\"key\\":\\"value\\"};
"
`;

exports[`Transform is code Type is "set" 1`] = `
"
  \\"use strict\\";
  Object.defineProperty(exports, \\"__esModule\\", {
      value: true
  });
  exports.default = {};
"
`;

exports[`Transform is code Type is "string" 1`] = `
"
  \\"use strict\\";
  Object.defineProperty(exports, \\"__esModule\\", {
      value: true
  });
  exports.default = \\"__string__\\";
"
`;

exports[`Transform is code Type is null 1`] = `
"
  \\"use strict\\";
  Object.defineProperty(exports, \\"__esModule\\", {
      value: true
  });
  exports.default = null;
"
`;

exports[`Transform is code Type is undefined 1`] = `
"
  \\"use strict\\";
  Object.defineProperty(exports, \\"__esModule\\", {
      value: true
  });
  exports.default = undefined;
"
`;
